### PR TITLE
Next + Typescript

### PR DESCRIPTION
**Pull request by vvscode**
https://nextjs.org/
http://typescriptlang.org

https://github.com/memory-cards/cards-exporter/issues/4
https://github.com/memory-cards/cards-exporter/issues/3

### CODE REVIEW

File: api/date/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    res = httpMocks.createResponse();
+  });
+
+  it(`should send object with date and apiName`, () => {
+    res.send = jest.fn();
+    dateController(req, res);
+
+    const sendParam = res.send.mock.calls[0][0];
+    expect(sendParam).toMatchObject({ apiName: "date" });
+    expect(sendParam).toHaveProperty("date");
```
[the remark this answers is not in the document]
Comment (vvscode) [reply api/date/index.spec.ts]: @zenby  
```
 FAIL  api/date/index.spec.ts
  dateController
    ✕ should send object with date and apiName (16ms)

  ● dateController › should send object with date and apiName

    expect(object).toHaveProperty(path)

    Expected the object:
      {"apiName": "date"}
    To have a nested property:
      "date"

      17 |     const sendParam = res.send.mock.calls[0][0];
      18 |     expect(sendParam).toMatchObject({ apiName: "date" });
    > 19 |     expect(sendParam).toHaveProperty("date");
         |                       ^
      20 |   });
      21 | });
      22 |

      at Object.it (api/date/index.spec.ts:19:23)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 failed, 2 passed, 3 total
```
is it clear what is wrong from that message?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    "lint:prettier": "prettier --list-different --parser typescript ./pages/**.{ts,tsx}",
+    "lint:tslint": "tslint --project .",
+    "lint": "yarn lint:prettier && yarn lint:tslint",
+    "precommit": "lint-staged",
+    "prettier": "prettier --parser typescript --write ./**/*.tsx",
+    "start:build": "NODE_ENV=production next build",
+    "start:server": "NODE_ENV=production ts-node server.ts",
+    "start": "yarn start:build && yarn start:server",
+    "test:jest": "jest",
+    "test": "yarn lint && yarn test:jest"
```
[the remark this answers is not in the document]
Comment (vvscode) [reply package.json]: do you mean `jest -w`  ?

File: api/date/index.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zenby) [reply api/date/index.spec.ts]: Exactly this thing. Thanks.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (zenby) [reply package.json]: Yes 
I've checked `-w` doesn't work but `--watch` works fine.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  "devDependencies": {
+    "@types/enzyme": "^3.1.14",
+    "@types/express": "^4.16.0",
+    "@types/jest": "^23.3.8",
+    "@types/next": "^7.0.3",
+    "@types/react": "^16.4.18",
+    "@types/react-test-renderer": "^16.0.3",
+    "enzyme-adapter-react-16": "^1.6.0",
+    "husky": "^1.1.2",
```
Comment (zenby) [inline package.json]: Vasiliy, do we have a setup for this library? Or it will be added later?

Comment (vvscode) [reply package.json]: it used for https://github.com/memory-cards/cards-exporter/pull/1/files/bea9a9410ae615582d0674241d264956e03a4849#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R16

it add precommit hook

Comment (zenby) [reply package.json]: I've found this dependency, thanks.

**Pull request merged**